codalab/rest/cli.py

```diff
@@ -0,0 +1,164 @@
+"""
+Web terminal API.
+"""
+from cStringIO import StringIO
+from datetime import datetime, timedelta
+import json
+from oauthlib.common import generate_token
+import shlex
+
+from bottle import (
+    httplib,
+    HTTPResponse,
+    local,
+    post,
+    request,
+    response,
+)
+
+from codalab.client.json_api_client import JsonApiClient
+from codalab.common import UsageError
+from codalab.lib import (
+    bundle_cli,
+    worksheet_util,
+)
+from codalab.lib.codalab_manager import CodaLabManager
+from codalab.objects.oauth2 import OAuth2Token
+
+
+@post('/cli/command')
+@post('/api/worksheets/command/')  # DEPRECATED ROUTE
+def post_worksheets_command():
+    # TODO(klopyrev): The Content-Type header is not set correctly in
+    # editable_field.jsx, so we can't use request.json.
+    data = json.loads(request.body.read())
+
+    if data.get('raw_command', None):
+        data['command'] = worksheet_util.get_worksheet_info_edit_command(data['raw_command'])
```

Comment:
This logic is kind of strange: why is raw_command only for edits?  The name wouldn't suggest that...